Lib/test/test_ast.py

```diff
@@ -582,6 +582,19 @@ def test_invalid_identifier(self):
             compile(m, "<test>", "exec")
         self.assertIn("identifier must be of type str", str(cm.exception))
 
+    def test_invalid_constant(self):
+        for invalid_constant in (int, (1, 2, int), frozenset((1, 2, int))):
```

Comment:
Just a nitpick: since you have so much parenthesis in the line, it may look clearer if you remove the outer one (they are not necessary):
```suggestion
        for invalid_constant in int, (1, 2, int), frozenset((1, 2, int)):
```